src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -430,7 +464,11 @@ private AcceptTxResult AddCore(Transaction tx, TxFilteringState state, bool isPe
                     previousTxBottleneck ??= tx.CalculateAffordableGasPrice(_specProvider.GetCurrentHeadSpec().IsEip1559Enabled,
                             _headInfo.CurrentBaseFee, balance);
 
-                    if (tx.Nonce == currentNonce + i)
+                    if (tx.MaxFeePerBlobGas < _headInfo.CurrentPricePerBlobGas)
+                    {
+                        gasBottleneck = UInt256.Zero;
+                    }
```

Comment:
Would this affect non-blob transactions?

Reply:
It is not affecting - for non-blob txs `MaxFeePerBlobGas` is null so check is skipped

Reply:
Could use a comment